src/main/java/uk/nhs/hee/trainee/details/api/TraineeProfileResource.java

```diff
@@ -48,37 +55,40 @@ public TraineeProfileResource(TraineeProfileService traineeProfileService,
     this.traineeProfileMapper = traineeProfileMapper;
   }
 
-
   /**
-   * Get a trainee profile record with a given ID.
+   * Get a trainee's profile.
    *
-   * @param traineeProfileId The ID of the trainee profile to get.
+   * @param token The authorization token from the request header.
    * @return The {@link PersonalDetailsDto} representing the trainee profile.
    */
-  @GetMapping("/trainee-profile/{id}")
-  public TraineeProfileDto getTraineeProfileById(
-      @PathVariable(name = "id") String traineeProfileId) {
-    log.trace("Trainee Profile of a trainee by traineeProfileId {}", traineeProfileId);
-    TraineeProfile traineeProfile = traineeProfileService.getTraineeProfile(traineeProfileId);
-    traineeProfile = traineeProfileService.hidePastProgrammes(traineeProfile);
-    traineeProfile = traineeProfileService.hidePastPlacements(traineeProfile);
-    return traineeProfileMapper.toDto(traineeProfile);
-  }
+  @GetMapping("/trainee-profile")
+  public ResponseEntity<TraineeProfileDto> getTraineeProfile(
+      @RequestHeader(HttpHeaders.AUTHORIZATION) String token) {
+    log.trace("Trainee Profile of authenticated user.");
+
+    String[] tokenSections = token.split("\\.");
+    byte[] payloadBytes = Base64.getDecoder()
+        .decode(tokenSections[1].getBytes(StandardCharsets.UTF_8));
+    String tisId;
+
+    try {
+      ObjectMapper mapper = new ObjectMapper();
```

Comment:
It's looks safe to declare this as a field and avoid creating it per-request.

Reply:
...or allowing Dependency Injection for testing?

Reply:
Good point, I'll inject it 👍